jdk/src/linux/native/jdk/internal/platform/cgroupv1/CgroupMetrics.c

```diff
@@ -33,3 +34,10 @@ Java_jdk_internal_platform_CgroupMetrics_isUseContainerSupport(JNIEnv *env, jcla
 {
     return JVM_IsUseContainerSupport();
 }
+
+JNIEXPORT jlong JNICALL
+Java_jdk_internal_platform_CgroupMetrics_getTotalMemorySize0
+  (JNIEnv *env, jclass ignored)
+{
+    return sysconf(_SC_PHYS_PAGES) * sysconf(_SC_PAGESIZE);
```

Comment:
According to my experiments changing this to:

```
    jlong page_size = sysconf(_SC_PAGESIZE);
    jlong pages = sysconf(_SC_PHYS_PAGES);
    return pages * page_size;
```

fixes the test issue on x86 (32 bit). As this is an issue with the upstream patch, I suggest to file an upstream bug to get it fixed as we'll likely have missed the issue on later JDKs (no testing happens on x86 AFAIK).

Reply:
Not that with an instrumented JDK I was able to observe the following (when the test fails):

```
systemMemory == -1308622848
containerMemory == 104857600
Metrics.getMemoryLimit() == -1
```

I.e. we were getting a negative host memory value.

Reply:
Thanks for working on this! I guess without the explicit intermediate variables some kind of integer promotion/demotion is going on and there's an overflow on 32 bit?

I'll raise the upstream bug, thanks!

Reply:
I wonder what the behaviour is on ARM32. I don't know whether we are testing that, whether it's explicitly supported for jdk8u or not, or by who. I've been wondering that since the reports of problems with non-static member initialization in another backport.

Reply:
Filed https://bugs.openjdk.org/browse/JDK-8300119

Reply:
I am happy to raise the upstream patch, or do you want to do it?

Reply:
Feel free to go ahead and post a patch for [JDK-8300119](https://bugs.openjdk.org/browse/JDK-8300119). Thanks! In this backport we could do `/issue add JDK-8300119` if you ask me.